packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -175,7 +230,12 @@ function GridFilterForm(props: GridFilterFormProps) {
     <GridFilterFormRoot className={classes.root}>
       <FormControl
         variant="standard"
-        sx={{ flexShrink: 0, justifyContent: 'flex-end', marginRight: 0.5, marginBottom: 0.2 }}
+        sx={[
+          { flexShrink: 0, justifyContent: 'flex-end', marginRight: 0.5, marginBottom: 0.2 },
+          ...(Array.isArray(deleteIconContainerSx)
+            ? deleteIconContainerSx
+            : [deleteIconContainerSx]),
+        ]}
```

Comment:
As we discussed, the prop could be renamed to `deleteIconContainerProps` and allow to override other props of the `FormControl`.

```suggestion
        {...deleteIconContainerProps}
        sx={{
          flexShrink: 0, 
          justifyContent: 'flex-end',
          marginRight: 0.5,
          marginBottom: 0.2,
          ...deleteIconContainerProps.sx
        }}
```

Another alternative is to make the `FormControl` a styled component, then you don't need to care abound spreading `deleteIconContainerProps.sx`, since it's done automatically.

```tsx
const StyledFormControl = styled(FormControl)({ flexShrink: 0 });

<StyledFormControl {...deleteIconContainerProps} />
```